pkg/endpoint/endpoint.go

```diff
@@ -2074,14 +2069,12 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 	if e.SecurityIdentity != nil {
 		oldIdentity := e.SecurityIdentity
 		defer func() {
-			lastUse, err := oldIdentity.Release()
+			err := oldIdentity.Release()
 			if err != nil {
 				elog.WithFields(logrus.Fields{logfields.Identity: oldIdentity.ID}).
 					WithError(err).Warn("BUG: Unable to release old endpoint identity")
 			}
-			if lastUse {
-				owner.RemoveNetworkPolicy(e.SecurityIdentity.ID)
-			}
+			owner.RemoveNetworkPolicy(e.IPv6, e.IPv4)
```

Comment:
Same as above. I don't think this call makes sense anymore. We should only call this when the endpoint is destroyed, which is not the case here.